aws_lambda_powertools/event_handler/api_gateway.py

```diff
@@ -713,10 +713,20 @@ def _remove_prefix(self, path: str) -> str:
             return path
 
         for prefix in self._strip_prefixes:
-            if path == prefix:
-                return "/"
-            if self._path_starts_with(path, prefix):
-                return path[len(prefix) :]
+            if isinstance(prefix, str):
+                if path == prefix:
+                    return "/"
+
+                if self._path_starts_with(path, prefix):
+                    return path[len(prefix) :]
+
+            if isinstance(prefix, Pattern):
+                path = re.sub(prefix, "", path)
+
+        # When using regexes, we might get into a point where everything is removed
+        # from the string, so we check if it's empty and change it accordingly.
+        if not path:
+            path = "/"
```

Comment:
nice catch!! should we warn? thinking of unintentional regex accidents or maybe it's intentional

Reply:
I've added a test and also moved the code so it's more consistent with the previous path.